### PR TITLE
Add ipamd changes for sg support

### PR DESCRIPTION
**Pull request by mogren**
*Issue #, if available:*
Follow up to #1125, will rebase once that PR is merged.

*Description of changes:*
* Add the needed tags on the node
* Read pod labels in `aws-node` and return the needed values to the CNI binary

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/ipamd/rpc_handler.go [hunk 1]
```diff
@@ -72,17 +146,44 @@ func (s *server) AddNetwork(ctx context.Context, in *rpc.AddNetworkRequest) (*rp
...
 func (s *server) DelNetwork(ctx context.Context, in *rpc.DelNetworkRequest) (*rpc.DelNetworkReply, error) {
 	log.Infof("Received DelNetwork for Sandbox %s", in.ContainerID)
 	delIPCnt.With(prometheus.Labels{"reason": in.Reason}).Inc()
-
+	if s.ipamContext.enablePodENI {
+		pod, err := s.ipamContext.GetPod(in.K8S_POD_NAME, in.K8S_POD_NAMESPACE)
+		if err != nil {
+			if k8serror.IsNotFound(err) {
+				log.Warn("Send AddNetworkReply: pod not found")
+				return &rpc.DelNetworkReply{Success: true}, nil
```
Comment (SaranBalaji90) [inline pkg/ipamd/rpc_handler.go:167]: Does CNI takes care if you send empty IP in the response?

File: pkg/networkutils/network_test.go [hunk 2]
```diff
@@ -563,10 +563,36 @@ func TestSetupHostNetworkIgnoringRpFilterUpdate(t *testing.T) {
 	setupNetLinkMocks(ctrl, mockNetLink)
 
 	var vpcCIDRs []string
-	err := ln.SetupHostNetwork(vpcCIDRs, loopback, &testENINetIP)
+	err := ln.SetupHostNetwork(vpcCIDRs, loopback, &testENINetIP, false)
 	assert.NoError(t, err)
 }
 
+//// TODO: fix test
```
Comment (SaranBalaji90) [inline pkg/networkutils/network_test.go:570]: This needs to be fixed?

File: pkg/ipamd/rpc_handler.go (hunk 1 above)
Comment (mogren) [reply pkg/ipamd/rpc_handler.go:167]: Yes, if we can't parse the IP returned, we just log a warning and exit. From the `del` function:
```
	deletedPodIP := net.ParseIP(r.IPv4Addr)
	if deletedPodIP != nil {
 .
 .
 .
	} else {
		log.Warnf("Container %s did not have a valid IP %s", args.ContainerID, r.IPv4Addr)
	}
	return nil
}
```

**Comment by mogren** [comment]
Note: Since the CircleCI tests run in my account, the results do not get reported correctly in the PR:

https://app.circleci.com/pipelines/github/mogren/amazon-vpc-cni-k8s/604/workflows/04c8c4e6-3699-4746-ba75-d624ab5dcde2

**Pull request merged**